docs/config.md

```diff
@@ -0,0 +1,51 @@
+# Configuration
+The philosophy we will use when exposing HCO configuration options is that only
+options that all operators consume can be exposed on the default HCO CR.
+Configuration options that are for specific component operators, will be hidden
+but accepted as valid configuration.
+
+Exposed variables are expressed in the HCO [CSV file](https://github.com/kubevirt/hyperconverged-cluster-operator/blob/dfdd4ac492c1d91c130eec03af1e4f6b04d54c7e/deploy/converged/olm-catalog/kubevirt-hyperconverged/0.0.1/kubevirt-hyperconverged-operator.v0.0.1.clusterserviceversion.yaml#L18-L20)
+```yaml
+          "spec": {
+            "CDIImagePullPolicy":"IfNotPresent",
+            "KubeVirtImagePullPolicy":"IfNotPresent"
+          }
+```
+
+## 1.0 Config Options
+All component operators must express each of the HCO `Config Options` on their CR.
+The component variable:
+  - does __not__ have to be the same name as the HCO variable
+  - can be expressed by multiple granular variables
+
+| Config Options | Description |
+|----|----|----|
+| Version  |  Product Version |
```

Comment:
Is this version of the operator or of the project it deploys?

In network addons we ship multiple projects, e.g. multiple versions. If we should support multiple versions of them, it would mean we have to keep definitions of all of them in the source code.

We don't support upgrades yet, but my idea was to ship one set of components in single operator release (e.g. network addons operator v0.3.0 would deploy multus X, bridge Y, upcoming operator version multus X+1, bridge Y+1).

Therefore upgrade would be done using SVC by redeploying the operator.

Reply:
This is the project version it deploys.  The HCO will distribute proper component versions to component CRs based on the Version.

The upgrade is triggered by changing the HCO version in the combined deployment.   We could change that to having the operator upgrade triggering the upgrade, but let's investigate at a later date.

Reply:
Yeah, let's talk about it later. I'd like to implement the upgrade first.

Reply:
@rthallisey Can't we just give you the full configured CR for a given product release version? We too deploy multiple different versions of components and the operator does not know or care about products using it.

Reply:
I think that's the solution David is proposing.